cli/flags.rs

```diff
@@ -100,6 +102,15 @@ fn add_run_args<'a, 'b>(app: App<'a, 'b>) -> App<'a, 'b> {
       Arg::with_name("no-fetch")
         .long("no-fetch")
         .help("Do not download remote modules"),
+    ).arg(
+      // TODO(mtharrison): Copied from Node for now - Discuss what the ideal flags are. Also include flag for host/port binding
+      Arg::with_name("inspect")
+        .long("inspect")
+        .help("Enable inspector"),
+    ).arg(
+      Arg::with_name("inspect-brk")
+        .long("inspect-brk")
+        .help("Enable inspector and pause on first statement")
```

Comment:
I think --debug is most reasonable. I'd like to avoid having two flags...

Reply:
Two flags are required for two different problems:
`--inspect` starts the process and allow the inspector to connect any time later, e.g. some tool would like to attach to a running program from time to time and capture one of the heap profilers to build aggregated report later,
`--inspect-brk` is for the clients which need to set up something ahead of time, e.g. set breakpoints or start profiler. In this mode inspected process should wait until inspector client is connected and sends `Runtime.runIfWaitingForDebugger` before run. At the same time, I am not sure that pause at first statement is an important part of this behavior.

If I would need to choose only one flag that covers most use cases, I will add `--inspect-brk` flag that waits for connected client and `Runtime.runIfWaitingForDebugger` protocol method without breakpoint at first statement.
DevTools or other clients can set required breakpoint manually - it should be possible to fetch entry file name after client connected and before it sent `Runtime.runIfWaitingForDebugger`

Reply:
Pause at first statement is essential for dedicated debugger frontends (e.g. the ones that only access source files from the debug target).

E.g. if you try to debug a short Node.js application using Chrome DevTools you would not be able to set a breakpoint as the application will run to a completion before you get a chance to interrupt it and set a breakpoint.

Reply:
I start the app with `--inspect-wait-for-debugger`, this flag waits until the client is connected and `Runtime.runIfWaitingForDebugger` is called.
My frontend connects to the app, calls something to fetch entry file name, e.g. `Runtime.evaluate({expression: 'process.entryFileName'})` after I send `Debugger.setBreakpointByUrl` and only after I prepared required breakpoint frontend calls `Runtime.runIfWaitingForDebugger`.

In this case, my frontend controls what and where breakpoints are set.

If we would like to have some magic for existing frontends then I prefer to have two separate flags: `--inspect-wait-for-debugger` and `--inspect-set-breakpoint-at-first-stmt`. Since we have some clients who do not need this breakpoint at first statement but would like to run some protocol command, e.g., `HeapProfiler.start`, before `Runtime.runIfWaitingForDebugger`.